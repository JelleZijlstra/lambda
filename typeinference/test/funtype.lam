# Checking that substitutions happen correctly on function types
data list a = Nil | Cons a (list a) in

let f = \x. Cons x Nil in
(f 3, f)
