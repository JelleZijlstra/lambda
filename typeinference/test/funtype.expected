%%stdout%%
Result: module
	let __result__ = (((Cons 3) Nil), \x. Cons x Nil)
end
%%stderr%%
